lib/internal/readline/interface.js

```diff
@@ -498,8 +510,22 @@ class Interface extends InterfaceConstructor {
     // erase data
     clearScreenDown(this.output);
 
-    // Write the prompt and the current buffer content.
-    this[kWriteToOutput](line);
+    // Check if this is a multiline entry
+    const lines = StringPrototypeSplit(this.line, '\n');
```

Comment:
Instead of checking this, could we have a state instead? I saw it's checked in many places, so having a symbol property that would know if we handle a multiline string or not would AFAIK just reduce the code overhead.